django/db/models/fields/json.py

```diff
@@ -0,0 +1,497 @@
+import json
+
+from django import forms
+from django.core import exceptions
+from django.db.models import lookups
+from django.db.models.lookups import (
+    FieldGetDbPrepValueMixin, Lookup, Transform,
+)
+from django.db.utils import NotSupportedError
+from django.utils.translation import gettext_lazy as _
+
+from . import Field
+from .mixins import CheckFieldDefaultMixin
+
+
+class JSONField(CheckFieldDefaultMixin, Field):
+    description = _('A JSON object')
+    default_error_messages = {
+        'invalid': _('Value must be valid JSON.'),
+    }
+    _default_hint = ('dict', '{}')
+
+    def __init__(
+        self, verbose_name=None, name=None, encoder=None,
+        decoder=None, *args, **kwargs
+    ):
+        self.encoder, self.decoder = encoder, decoder
+        super().__init__(verbose_name, name, *args, **kwargs)
+
+    def check(self, **kwargs):
+        return super().check(**kwargs)
```

Comment:
You should also be able to drop this whole method now @felixmm.

Reply:
Not really, because we want to call `check()` from `CheckFieldDefaultMixin`.

Reply:
It would be called anyway thanks to inheritance?

Reply:
Yes, right. :man_facepalming:

:black_circle: 🕳️

Reply:
> You should also be able to drop this whole method now @felixmm.

Alright, this has gone for long enough. Please, whenever you tag @felixxm check that it is TWO x and ONE m, not @felixmm (me) with ONE x and TWO m

Reply:
> Alright, this has gone for long enough.

Sorry. I'm pretty sure I normally get this right - this was an accidental typo or a tedious phone autocorrection. It is just unfortunate that you happen to have a very similar username...